src/gwt/src/org/rstudio/studio/client/common/mathjax/MathJax.java

```diff
@@ -334,23 +352,54 @@ public void execute()
             mathjaxTypeset(el, text, new MathJaxTypesetCallback()
             {
                @Override
-               public void onMathJaxTypesetComplete(boolean error)
+               public void onMathJaxTypesetComplete(final boolean error)
                {
-                  // re-position the element
-                  int height = el.getOffsetHeight() + 30;
-                  Element ppElement = el.getParentElement().getParentElement();
-                  ppElement.getStyle().setHeight(height, Unit.PX);
-                  docDisplay_.onLineWidgetChanged(lineWidget);
-                  
-                  // invoke supplied callback
-                  if (callback != null)
-                     callback.onMathJaxTypesetComplete(error);
+                  // force expansion
+                  withExpandedLineWidget(lineWidget, new CommandWithArg<Boolean>()
+                  {
+                     @Override
+                     public void execute(Boolean stateChanged)
+                     {
+                        // re-position the element
+                        int height = el.getOffsetHeight() + 30;
+                        Element ppElement = el.getParentElement().getParentElement();
+                        ppElement.getStyle().setHeight(height, Unit.PX);
+                        docDisplay_.onLineWidgetChanged(lineWidget);
+
+                        // invoke supplied callback
+                        if (callback != null)
+                           callback.onMathJaxTypesetComplete(error);
+                     }
+                  });
                }
             });
          }
       });
    }
    
+   private boolean isLineWidgetCollapsed(int row)
+   {
+      LineWidget widget = docDisplay_.getLineWidgetForRow(row);
+      if (widget == null)
+         return false;
+      
+      ChunkOutputWidget cow = lwToPlwMap_.get(widget);
+      if (cow == null)
+         return false;
+      
+      return cow.getExpansionState() == ChunkOutputWidget.COLLAPSED;
+   }
+   
+   private void withExpandedLineWidget(LineWidget widget,
+                                       final CommandWithArg<Boolean> onExpansionCompleted)
+   {
+      ChunkOutputWidget cow = lwToPlwMap_.get(widget);
+      if (cow == null)
+         return;
```

Comment:
realize we never expect this to happen, but perhaps it'd be more fault tolerant to `onExpansionCompleted(false)` here so that whatever the caller wanted to do gets done anyway?